src/internal/document_api/mod.rs

```diff
@@ -522,28 +708,148 @@ fn encrypt_document_core<'a, CR: rand::CryptoRng + rand::RngCore>(
+    fn from(edek: &EncryptedDek) -> Self {
+        use crate::proto::edeks;
+        use recrypt::api as re;
+        let edek = edek.clone();
+
+        // transform the recrypt EncryptedValue to a edek proto
+        let mut proto_edek_data_no_pubk = match edek.encrypted_dek_data {
+            re::EncryptedValue::EncryptedOnceValue {
+                ephemeral_public_key,
+                encrypted_message,
+                auth_hash,
+                public_signing_key,
+                signature,
+            } => {
+                let mut proto_edek_data = EncryptedDekDataP::default();
+                {
+                    let mut proto_eph_pub_key = edeks::PublicKey::default();
+                    let (x, y) = ephemeral_public_key.bytes_x_y();
+                    proto_eph_pub_key.set_x(x[..].into());
+                    proto_eph_pub_key.set_y(y[..].into());
+                    proto_edek_data.set_ephemeralPublicKey(proto_eph_pub_key);
+                }
+                proto_edek_data.set_encryptedMessage(encrypted_message.bytes()[..].into());
+                proto_edek_data.set_signature(signature.bytes()[..].into());
+                proto_edek_data.set_authHash(auth_hash.bytes()[..].into());
+                proto_edek_data.set_publicSigningKey(public_signing_key.bytes()[..].into());
+
+                proto_edek_data
+            }
+            re::EncryptedValue::TransformedValue { .. } => {
+                unreachable!("Will be needed for decrypt!")
+            }
+        };
+
+        //convert the grants
+        let (proto_uog, proto_pub_key) = match edek.grant_to {
+            WithKey {
+                id:
+                    UserOrGroup::User {
+                        id: UserId(user_string),
+                    },
+                public_key,
+            } => {
+                let mut proto_uog = edeks::UserOrGroup::default();
+                proto_uog.set_userId(user_string.into());
+
+                let mut proto_pub_key = edeks::PublicKey::default();
+                proto_pub_key.set_x(public_key.as_bytes().into());
+                proto_pub_key.set_y(public_key.as_bytes().into());
+                (proto_uog, proto_pub_key)
+            }
+            WithKey {
+                id:
+                    UserOrGroup::Group {
+                        id: GroupId(group_string),
+                    },
+                public_key,
+            } => {
+                let mut proto_uog = edeks::UserOrGroup::default();
+                proto_uog.set_groupId(group_string.into());
+
+                let mut proto_pub_key = edeks::PublicKey::default();
+                proto_pub_key.set_x(public_key.as_bytes().into());
+                proto_pub_key.set_y(public_key.as_bytes().into());
+                (proto_uog, proto_pub_key)
+            }
+        };
+
+        // attach the ephemeral public key
+        proto_edek_data_no_pubk.set_ephemeralPublicKey(proto_pub_key);
```

Comment:
I don't think this is right. Why are you setting the ephemeral public key to a public key you pulled off a user or group?

Reply:
You are correct. This function has been refactored a few times (and just was again), and something got messed up along the way. Thanks!